components/misc_metrics/menu_metrics.h

```diff
@@ -33,10 +33,21 @@ class MenuMetrics {
 
   void RecordMenuGroupAction(MenuGroup group);
 
+  void RecordMenuShown();
+  void RecordMenuDismiss();
+
  private:
+  void RecordMenuDismissRate();
+
+  void Update();
+
   base::flat_map<MenuGroup, double> menu_group_access_counts_;
 
   raw_ptr<PrefService> local_state_ = nullptr;
+  WeeklyStorage menu_shown_storage_;
+  WeeklyStorage menu_dismiss_storage_;
+
+  base::RepeatingTimer update_timer_;
```

Comment:
you need `WallClockTimer` or a smaller interval

Reply:
can you pls elaborate on why a wall clock timer would be needed here?

Reply:
it seems to me that it is always needed when the interval is relatively large, say more than an hour or two? apparently most of devices can't avoid sleeping during a day

Reply:
good point. there are quite a few places where we use repeating timers with daily intervals (i.e. wallet). we should revisit those cases at some point